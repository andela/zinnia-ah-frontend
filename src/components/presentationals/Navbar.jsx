import React from 'react';
import PropTypes from 'prop-types';

import { Dropdown } from 'semantic-ui-react';
import Avatar from './Avatar.jsx';
import './Avatar.jsx';
import './Navbar.scss';

const Navbar = ({ props }) => {
  return (
    <div>
      <div className="navbar">
        <div>
          <h1 className="brand">Authors Haven</h1>
        </div>
        {props.profileUrl !== '' && (
          <div className="navbar-avatar">
            <Avatar avatarUrl={props.profileUrl} />
            <Dropdown className="dropdown">
              <Dropdown.Menu>
                <Dropdown.Item>
                  <a>Logout</a>
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </div>
        )}
      </div>
    </div>
  );
};

Navbar.propTypes = {
  props: PropTypes.shape({
    profileUrl: PropTypes.string,
  }),
  profileUrl: PropTypes.string,
};

export default Navbar;
