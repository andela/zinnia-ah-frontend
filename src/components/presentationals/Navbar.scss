@import url("https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento");
@import "../../assets/variables";

.navbar {
  align-items: center;
  display: flex;
  height: 100px;
  justify-content: space-between;
  padding: 0% 6%;

  .brand {
    font-family: "Sacramento", cursive;
    font-size: 36px;
    line-height: 53px;
  }

  .navbar-avatar {
    align-items: center;
    display: flex;
  }
}
