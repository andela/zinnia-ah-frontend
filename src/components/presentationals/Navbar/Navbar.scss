@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento');
@import '../../../assets/variables';

.navbar {
  align-items: center;
  display: flex;
  height: 6.25rem;
  justify-content: space-between;
  padding: 0% 6%;

  .brand {
    color: $primary-black;
    font-family: 'Sacramento', cursive;
    font-size: 2.25rem;
    line-height: 3.3125rem;
  }

  .navbar-avatar {
    align-items: center;
    display: flex;
  }
}

.sm-hide {
  display: block;
}

@media only screen and (max-width: 56.25rem) {
  .navbar {
    justify-content: center;
  }

  .sm-hide {
    display: none;
  }
}
