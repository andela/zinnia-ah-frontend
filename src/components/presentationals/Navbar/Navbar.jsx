import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dropdown } from 'semantic-ui-react';

// stylesheets
import './Navbar.scss';

// modules
import { logout, autoLogin } from '../../../store/modules/auth';

// component
import Avatar from '../Avatar/Avatar';
import Loader from '../Loader/Loader';

// images
import { DEFAULT_PROFILE_PICTURE } from '../../../config/config';

export class Navbar extends Component {
  componentDidMount() {
    this.props.autoLogin();
  }

  logOut = event => {
    event.preventDefault();
    this.props.logout();
  };
  render() {
    const { className, children, isLoading, user } = this.props;
    return (
      <div>
        {isLoading && <Loader text="loading. please wait" size="large" />}
        <div className="navbar">
          <div>
            <Link to="/">
              <h1 className="brand">Authors Haven</h1>
            </Link>
          </div>
          <div
            style={{
              display: 'flex',
              justifyContent: 'flex-end',
              flexDirection: 'row-reverse',
            }}
          >
            {children && user && (
              <Fragment>
                <div className="navbar-avatar" style={{ margin: '0 2rem' }}>
                  <Avatar
                    className={className}
                    url={user.image || DEFAULT_PROFILE_PICTURE}
                  />
                  <Dropdown className="dropdown">
                    <Dropdown.Menu>
                      <Dropdown.Item>
                        <Link to={`@${user.username}`}>Profile</Link>
                      </Dropdown.Item>
                      <Dropdown.Item>
                        <a onClick={this.logOut}>Logout</a>
                      </Dropdown.Item>
                    </Dropdown.Menu>
                  </Dropdown>
                </div>
              </Fragment>
            )}
            {children}
          </div>
        </div>
      </div>
    );
  }
}

Navbar.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
  url: PropTypes.string,
  logout: PropTypes.func,
  isLoading: PropTypes.bool,
  autoLogin: PropTypes.func,
  user: PropTypes.object,
};
const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  isAuthenticated: state.auth.loggedInUser !== null,
  user: state.auth.loggedInUser,
});

export default connect(
  mapStateToProps,
  { logout, autoLogin },
)(Navbar);
