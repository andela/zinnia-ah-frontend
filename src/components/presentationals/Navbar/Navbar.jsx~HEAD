import React from 'react';
import PropTypes from 'prop-types';

import { Dropdown } from 'semantic-ui-react';
import Avatar from '../Avatar/Avatar';
import './Navbar.scss';

const Navbar = ({ profileUrl }) => {
  return (
    <div>
      <div className="navbar">
        <div>
          <h1 className="brand">Authors Haven</h1>
        </div>
        {profileUrl && (
          <div className="navbar-avatar">
            <Avatar avatarUrl={profileUrl} />
            <Dropdown className="dropdown">
              <Dropdown.Menu>
                <Dropdown.Item>
                  <a>Logout</a>
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </div>
        )}
      </div>
    </div>
  );
};

Navbar.propTypes = {
  props: PropTypes.shape({
    profileUrl: PropTypes.string,
  }),
  profileUrl: PropTypes.string,
};

export default Navbar;
