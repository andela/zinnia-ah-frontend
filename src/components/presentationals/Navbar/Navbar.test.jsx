import React from 'react';
import { shallow } from 'enzyme';

import { Navbar } from './Navbar';

describe('Navbar', () => {
  const props = { autoLogin: jest.fn() };
  it('should render correctly', () => {
    const component = shallow(
      <Navbar {...props} profileUrl="http://dummyimage.com" className="" />,
    );
    expect(component).toMatchSnapshot();
  });
});
