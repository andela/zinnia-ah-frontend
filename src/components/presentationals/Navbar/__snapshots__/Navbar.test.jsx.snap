// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<div>
  <div
    className="navbar"
  >
    <div>
      <Link
        to="/"
      >
        <h1
          className="brand"
        >
          Authors Haven
        </h1>
      </Link>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
          "justifyContent": "flex-end",
        }
      }
    />
  </div>
</div>
`;
